src/Analysis/GWASResultsV2/Views/Results/Results.jsx

```diff
@@ -1,16 +1,150 @@
-import React, { useContext } from 'react';
+import React, { useContext, useState } from 'react';
+import { useQuery } from 'react-query';
+import { Spin, Button } from 'antd';
+
 import DetailPageHeader from '../../SharedComponents/DetailPageHeader/DetailPageHeader';
 import SharedContext from '../../Utils/SharedContext';
+import {
+  fetchPresignedUrlForWorkflowArtifact,
+  queryConfig,
+} from '../../Utils/gwasWorkflowApi';
+import '../../../GWASApp/GWASApp.css';
 
 const Results = () => {
+  const [imageLoaded, setImageLoaded] = useState(false);
+  const [imageLoadFailed, setImageLoadFailed] = useState(false);
   const { selectedRowData } = useContext(SharedContext);
   const { name, uid } = selectedRowData;
+
+  const { data, status } = useQuery(
+    [
+      'fetchPresignedUrlForWorkflowArtifact',
+      name,
+      uid,
+      'manhattan_plot_index',
+    ],
+    () => fetchPresignedUrlForWorkflowArtifact(
+      name,
+      uid,
+      'manhattan_plot_index',
+    ),
+    queryConfig,
+  );
+
+  const displayError = (errorMessage) => (
+    <div>
+        ❌ {errorMessage}
+    </div>
+  );
+
+  const downloadAll = () => {
+    fetchPresignedUrlForWorkflowArtifact(
+      name,
+      uid,
+      'gwas_archive_index',
+    ).then((res) => {
+      window.open(res, '_blank');
+    }).catch((error) => {
+      alert(`Could not download. \n\n${error}`);
+    });
+  };
+
+  const downloadManhattanPlot = () => {
+    fetchPresignedUrlForWorkflowArtifact(
+      name,
+      uid,
+      'manhattan_plot_index',
+    ).then((res) => {
+      window.open(res, '_blank');
+    }).catch((error) => {
+      alert(`Could not download. \n\n${error}`);
+    });
+  };
+
+  const displayTopSection = () => (
+    <React.Fragment>
+      <div className='GWASUI-flexRow' style={{ width: '100%' }}>
+        <div className='GWASUI-flexCol'>
+          <DetailPageHeader pageTitle={'Results'} />
+        </div>
+        <div className='GWASUI-flexCol'>
+          <Button
+            style={{ width: '50%' }}
+            onClick={downloadAll}
+          >Download All Results
+          </Button>
+        </div>
+      </div>
+      <div className='GWASUI-flexRow' style={{ background: 'rgb(209,219,229)', width: '100%' }}>
+        <div className='GWASUI-flexCol' todo='this is for the QQ plot button' />
+        <div className='GWASUI-flexCol'>
+          <Button
+            style={{ width: '50%' }}
+            onClick={downloadManhattanPlot}
+          >Download Manhattan Plot
```

Comment:
I'm thinking there is a way to make this image download for the end user instead of opening the image in a new tab (likely by generating a hyperlink with a ```download``` attribute), what do you think of me creating a ticket to work on that?

Reply:
Per our discussion: this is potentially difficult.

Reply:
Yes...the easier solutions only work if the image is served in the same domain, which is not the case in our app.